deeplearning4j/deeplearning4j-dataimport-solrj/src/test/java/org/deeplearning4j/nn/dataimport/solr/client/solrj/io/stream/TupleStreamDataSetIteratorTest.java

```diff
@@ -0,0 +1,188 @@
+    updateRequest.commit(cluster.getSolrClient(), "mySolrCollection");
+  }
+
+  private static class CountingIterationListener extends ScoreIterationListener {
+
+    private int numIterationsDone = 0;
+
+    public CountingIterationListener() {
+      super(1);
+    }
+
+    public int numIterationsDone() {
+      return numIterationsDone;
+    }
+
+    @Override
+    public void iterationDone(Model model, int iteration, int epoch) {
+      super.iterationDone(model, iteration, epoch);
+      ++numIterationsDone;
+    }
+
+  }
+
+  @Test
+  public void iterateTest() throws Exception {
+    doIterateTest(true);
+    doIterateTest(false);
+  }
+
+  private void doIterateTest(boolean withIdKey) throws Exception {
+
+      try (final TupleStreamDataSetIterator
+      tsdsi = new TupleStreamDataSetIterator(
+        123 /* batch */,
+        (withIdKey ? "greeting" : null) /* idKey */,
+        new String[] { "pie" },
+        new String[] { "answer" },
+        "tuple(greeting=\"hello world\",pie=3.14,answer=42)",
+        null)) {
+
+      assertTrue(tsdsi.hasNext());
+      final DataSet ds = tsdsi.next();
+
+      assertEquals(1, ds.getFeatures().length());
+      assertEquals(3.14f, ds.getFeatures().getFloat(0), 0.0f);
+
+      assertEquals(1, ds.getLabels().length());
+      assertEquals(42f, ds.getLabels().getFloat(0), 0.0f);
+
+      assertFalse(tsdsi.hasNext());
+    }
+  }
+
+  @Test
+  public void modelFitTest() throws Exception {
+
+    final MultiLayerNetwork model = new MultiLayerNetwork(
+      new NeuralNetConfiguration.Builder()
+        .list(
+          new OutputLayer.Builder(LossFunction.MSE)
+            .nIn(3)
+            .nOut(1)
+            .weightInit(WeightInit.ONES)
+            .activation(Activation.IDENTITY)
+            .build()
+          )
+        .pretrain(false)
+        .backprop(true)
+        .build()
+      );
+    model.init();
+
+    int batch = 1;
+    for (int ii=1; ii<=5; ++ii) {
+      final CountingIterationListener listener = new CountingIterationListener();
+      model.setListeners(listener);
+      batch *= 2;
+
+      try (final TupleStreamDataSetIterator tsdsi =
+          new TupleStreamDataSetIterator(
+            batch,
+            "id" /* idKey */,
+            new String[] { "channel_b_f", "channel_g_f", "channel_r_f" },
+            new String[] { "luminance_f" },
+            "search(mySolrCollection," +
+            "q=\"id:*\"," +
+            "fl=\"id,channel_b_f,channel_g_f,channel_r_f,luminance_f\"," +
+            "sort=\"id asc\"," +
+            "qt=\"/export\")",
+            cluster.getZkClient().getZkServerAddress())) {
+
+        model.fit(tsdsi);
+      }
+
+      assertEquals("numIterationsDone="+listener.numIterationsDone()+" numDocs="+numDocs+" batch="+batch,
+                   (numDocs+(batch-1))/batch, listener.numIterationsDone());
+    }
+
+    Nd4j.getWorkspaceManager().destroyAllWorkspacesForCurrentThread(); // TODO: seems to have no effect?
```

Comment:
One TODO item left here. Would appreciate guidance on how to proceed -- is anyone able to reproduce locally or could it be something environment in my specific setup? Is there perhaps some other destroy or cleanup API that should be called instead of this one here? Thanks!